scipy/cluster/tests/test_vq.py

```diff
@@ -250,10 +250,19 @@ def test_kmeans2_init(self):
         kmeans2(data, 3, minit='points')
         kmeans2(data[:, :1], 3, minit='points')  # special case (1-D)
 
-        kmeans2(data, 3, minit='random')
-        kmeans2(data[:, :1], 3, minit='random')  # special case (1-D)
+        kmeans2(data, 3, minit='++')
+        kmeans2(data[:, :1], 3, minit='++')  # special case (1-D)
 
-    @pytest.mark.skipif(sys.platform == 'win32', reason='Fails with MemoryError in Wine.')
+        # minit='random' can give warnings, filter those
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                    'ignore',
+                    message="At least one cluster is empty. Re-run")
```

Comment:
This causes a test failure:
```
warning_calls = (['cluster/tests/test_vq.py:258', 'optimize/optimize.py:814', 'sparse/__init__.py:248'], ['__init__.py:113', 'cluster/hierarchy.py:3547', 'cluster/vq.py:138', 'cluster/vq.py:579', 'cluster/vq.py:695', 'constants/codata.py:1196', ...])

    @pytest.mark.slow
    @pytest.mark.skipif(sys.version_info < (3, 4), reason="needs Python >= 3.4")
    def test_warning_calls_filters(warning_calls):
        bad_filters, bad_stacklevels = warning_calls
    
        # There is still one simplefilter occurrence in optimize.py that could be removed.
        bad_filters = [item for item in bad_filters
                       if 'optimize.py' not in item]
        # The filterwarnings call in sparse/__init__.py is needed.
        bad_filters = [item for item in bad_filters
                       if os.path.join('sparse', '__init__.py') not in item]
    
        if bad_filters:
            raise AssertionError(
                "warning ignore filter should not be used, instead, use\n"
                "scipy._lib._numpy_compat.suppress_warnings (in tests only);\n"
                "found in:\n    {}".format(
>                   "\n    ".join(bad_filters)))
E           AssertionError: warning ignore filter should not be used, instead, use
E           scipy._lib._numpy_compat.suppress_warnings (in tests only);
E           found in:
E               cluster/tests/test_vq.py:258

bad_filters = ['cluster/tests/test_vq.py:258']
bad_stacklevels = ['__init__.py:113', 'cluster/hierarchy.py:3547', 'cluster/vq.py:138', 'cluster/vq.py:579', 'cluster/vq.py:695', 'constants/codata.py:1196', ...]
warning_calls = (['cluster/tests/test_vq.py:258', 'optimize/optimize.py:814', 'sparse/__init__.py:248'], ['__init__.py:113', 'cluster/hierarchy.py:3547', 'cluster/vq.py:138', 'cluster/vq.py:579', 'cluster/vq.py:695', 'constants/codata.py:1196', ...])
```

You can fix it by using `suppress_warnings` from `scipy._lib._numpy_compat`. There's one other example of `suppress_warnings` usage in this file